src/overlays/actors/ovl_Bg_Ikninside/z_bg_ikninside.c

```diff
@@ -28,30 +32,125 @@ const ActorInit Bg_Ikninside_InitVars = {
+    this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, colHeader);
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->dyna.actor, &sCylinderInit);
+    Collider_UpdateCylinder(&this->dyna.actor, &this->collider);
+    if (Flags_GetSwitch(globalCtx, DMIKNINSIDE_GET_SWITCH(thisx) >> 9)) {
+        Actor_MarkForDeath(&this->dyna.actor);
+    }
+}
+
+void BgIkninside_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgIkninside* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
+
+void func_80C07220(BgIkninside* this, GlobalContext* globalCtx) {
+}
+
+void func_80C07230(BgIkninside* this, GlobalContext* globalCtx) {
+    if (this->dyna.actor.cutscene == -1) {
+        this->actionFunc = func_80C07220;
+    } else if (ActorCutscene_GetCurrentIndex() == 124) {
+        ActorCutscene_Stop(124);
+        ActorCutscene_SetIntentToPlay(this->dyna.actor.cutscene);
+    } else if (ActorCutscene_GetCanPlayNext(this->dyna.actor.cutscene)) {
+        ActorCutscene_StartAndSetUnkLinkFields(this->dyna.actor.cutscene, &this->dyna.actor);
+        this->actionFunc = func_80C07220;
+    } else {
+        ActorCutscene_SetIntentToPlay(this->dyna.actor.cutscene);
+    }
+}
+
+void func_80C072D0(BgIkninside* this, GlobalContext* globalCtx) {
+    s16 altitude;
+    s16 azimuth;
+    Vec3f pos;
+    Vec3f velocity;
+    f32 speed;
+    s32 i;
+
+    if (this->collider.base.acFlags & 2) {
+        if ((this->collider.info.acHitInfo != NULL) && (this->collider.info.acHitInfo->toucher.dmgFlags & 0x80000000)) {
+            for (i = 0; i < 20; i++) {
+                altitude = Rand_S16Offset(0x1800, 0x2800);
+                azimuth = ((u32)Rand_Next() >> 0x10);
+                speed = Rand_ZeroFloat(3.0f) + 8.0f;
+                velocity.x = speed * Math_CosS(altitude) * Math_SinS(azimuth);
+                velocity.y = speed * Math_SinS(altitude) + Rand_ZeroFloat(5.0f);
+                velocity.z = speed * Math_CosS(altitude) * Math_CosS(azimuth);
+                pos.x = Rand_ZeroFloat(10.0f) * velocity.x + this->dyna.actor.world.pos.x;
+                pos.y = Rand_ZeroFloat(1.0f) * velocity.y + this->dyna.actor.world.pos.y;
+                pos.z = Rand_ZeroFloat(10.0f) * velocity.z + this->dyna.actor.world.pos.z;
+                EffectSsHahen_Spawn(globalCtx, &pos, &velocity, &D_80C076D4, 0, 30, 566, 25, D_80C076A0[i & 1]);
+            }
+            Flags_SetSwitch(globalCtx, DMIKNINSIDE_GET_SWITCH(&this->dyna.actor) >> 9);
+            this->actionFunc = func_80C07230;
+            this->dyna.actor.draw = NULL;
+            func_800C62BC(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+        } else {
+            this->unk_1AA = 20;
+        }
+    }
+    if (this->unk_1AA > 0) {
+        if (this->unk_1AA & 1) {
```

Comment:
```suggestion
        if ((this->unk_1AA % 2) != 0) {
```
Out of curiosity does this work?

Reply:
yes, it does, do you want it to be like this? why?

Reply:
I think it would be preferable yes.
`this->unk_1AA` doesn't look to contain flags, but instead seems to contain a timer, note how it decrements later on. So it seems better to show the operation more as `is the number not a multiple of 2` vs `is the last bit set`.